Apply standard and cleanup of Measure-DbaDiskSpaceRequirement

wsmelton opened the pull request:
General cleanup of standards.

Addressed variable and properties names to make less generic and a bit clearer in reading output.

Removed use of EnableException parameter on Write-Message and Stop-Function calls.

Added code block for building connections to source and destination, to allow it to also properly test the access to each before trying to run the code.

@FriedrichWeinmann and/or @potatoqualitee  please review...if there is anything that needs to be addressed let me know (or by all means address, the branch is in the dbatools repo).

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
         [string]$DestinationDatabase,
         [PSCredential]$DestinationSqlCredential,
         [PSCredential]$Credential
```
Needs EnableException added, as well as docs

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -24,20 +23,27 @@ Function Measure-DbaDiskSpaceRequirement {
 
         .PARAMETER Credential
-            Windows credentials to connect via CIM/WMI/PowerShell remoting for MountPoint definition.
+            The credentials to use to connect via CIM/WMI/PowerShell remoting.
```
Add Docs for EnableException

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
         [PSCredential]$Credential
     )
     begin {
-        $NullText = '#NULL'
```
this should just be $null in the output - I'm not aware of any value in #NULL

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
         [DbaInstanceParameter]$Source,
-        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
         [string]$SourceDatabase,
```
Please change to just Database - I did a mass replace for SourceDatabase -> Database and it worked well, including docs

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
-                })
+            if (!$found) {
+                # Files on source but not on destination
+                $details += @([PSCustomObject]@{
```
this should be released to the pipeline immediately and not collected

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
-        $Detail | Select-DefaultView -Property DatabaseName1, DatabaseName2, Name1, Name2, SizeKB1, SizeKB2, DiffKB, ComputerName, MountPoint
+        if (Test-FunctionInterrupt) { return }
+
+        $details | Select-DefaultView -Property DatabaseName1, DatabaseName2, Name1, Name2, SizeKB1, SizeKB2, DiffKB, ComputerName, MountPoint
```
Unfortunately, I didn't see much in the output for this PR. Something went wrong somewhere.

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
     }
     end {
-        $Detail | Select-DefaultView -Property DatabaseName1, DatabaseName2, Name1, Name2, SizeKB1, SizeKB2, DiffKB, ComputerName, MountPoint
+        if (Test-FunctionInterrupt) { return }
```
Not required at this stage

potatoqualitee commented on functions/Measure-DbaDiskSpaceRequirement.ps1:
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
+                            DestinationLogicalName  = $destFile.Name
+                            SourceFileName          = $sourceFile.FileName
+                            DestinationFileName     = $destFile.FileName
+                            SourceFileSizeKB        = $sourceFile.Size
```
Can we use [dbasize] here?

potatoqualitee replied:
And anywhere else relevant. may require changing to bytes

wsmelton replied:
This is just returning the raw value, which is in KB. I can see how the DB space command handles it and just match that.

wsmelton commented:
@potatoqualitee if you get a chance give this one a spin, fixed the output issue as I had to update properties on the call to default view...forgot to update those.

I added DbaSize and think I did this right, the size unit from SMO is in KB so I simply had it product of that value times 1k and that should be the value in Bytes.

potatoqualitee commented:
sweet will do, thank you shawn. sorry for the delay, the past 3 days have been crazy!

potatoqualitee commented:
LOVE! such a cool command. thanks for the updates @wsmelton and thanks for the original command @pollusb !!

![image](https://user-images.githubusercontent.com/8278033/36340241-14e740f6-13d9-11e8-83be-a4f1406d134a.png)
